internal/ringbuffer.go

```diff
@@ -0,0 +1,119 @@
+	parSpecProxy, err := newPartitionSpecificProxy(client, serviceName, name)
+	if err != nil {
+		return nil, err
+	}
+	return &RingbufferProxy{parSpecProxy}, nil
+}
+
+func (rp *RingbufferProxy) Capacity() (capacity int64, err error) {
+	request := RingbufferCapacityEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferCapacityDecodeResponse)
+}
+func (rp *RingbufferProxy) Size() (size int64, err error) {
+	request := RingbufferSizeEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferSizeDecodeResponse)
+}
+
+func (rp *RingbufferProxy) TailSequence() (tailSequence int64, err error) {
+	request := RingbufferTailSequenceEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferTailSequenceDecodeResponse)
+}
+
+func (rp *RingbufferProxy) HeadSequence() (headSequence int64, err error) {
+	request := RingbufferHeadSequenceEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferHeadSequenceDecodeResponse)
+}
+
+func (rp *RingbufferProxy) RemainingCapacity() (remainingCapacity int64, err error) {
+	request := RingbufferRemainingCapacityEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferRemainingCapacityDecodeResponse)
+}
+
+func (rp *RingbufferProxy) Add(item interface{}, overflowPolicy core.OverflowPolicy) (sequence int64, err error) {
+	itemData, err := rp.validateAndSerialize(item)
+	if err != nil {
+		return
+	}
+	request := RingbufferAddEncodeRequest(rp.name, int32(overflowPolicy.Policy()), itemData)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferAddDecodeResponse)
+}
+
+func (rp *RingbufferProxy) AddAll(items []interface{}, overflowPolicy core.OverflowPolicy) (lastSequence int64, err error) {
+	itemsData, err := rp.validateAndSerializeSlice(items)
+	if err != nil {
+		return
+	}
+	request := RingbufferAddAllEncodeRequest(rp.name, itemsData, int32(overflowPolicy.Policy()))
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferAddAllDecodeResponse)
+}
+
+func (rp *RingbufferProxy) ReadOne(sequence int64) (item interface{}, err error) {
+	if err = rp.checkSequence(sequence, "sequence"); err != nil {
+		return
+	}
+	request := RingbufferReadOneEncodeRequest(rp.name, sequence)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToObjectAndError(responseMessage, err, RingbufferReadOneDecodeResponse)
+}
+
+func (rp *RingbufferProxy) ReadMany(startSequence int64, minCount int32, maxCount int32, filter interface{}) (items []interface{}, readCount int32, itemSequences []int64, err error) {
```

Comment:
There is no sync api for ReadMany in other clients.

Reply:
go is more like python. You can see each method both sync and async.